### ISSUE TITLE
Find all occurances - make it bold

### ISSUE THREAD

**Issue by Noitarud** [AndBible/and-bible#2225]
I'm always frustrated when: translators write various words for a specific foreign word (to the point that it seems they have no idea⁸) when I select Find all occurances I do not always know what the word is representing so can you make that part bold?

⁸KJV eg in the ten commandments. Vain. In a few verses that it occurs in but not translated as vain, they have another Hebrew word they tanslate as vain, and when you check that (try Zec10:2), it is translated as vain in every place. And Oppress* could be representing any one of 25 Hebrew words. I am frustrated to the point of wanting to do my own translating.

**Comment by timbze** [comment]
This feature has been implemented in alpha version. You have to manually install the apk if you want to try it. See https://github.com/AndBible/and-bible/releases/tag/alpha-650 and download the apk file under assets.

**Comment by timbze** [comment]
@Noitarud do you speak more than 1 language? I'm asking because I do and there's many words that you don't translate the same in every context. Languages are different and the Hebrew word for vain translating as "vain" or "false" in different contexts is not surprising to me.

**Comment by agrogers** [comment]
> ⁸KJV eg in the ten commandments. Vain. In a few verses that it occurs in but not translated as vain, they have another Hebrew word they tanslate as vain, and when you check that (try Zec10:2), it is translated as vain in every place. And Oppress* could be representing any one of 25 Hebrew words. I am frustrated to the point of wanting to do my own translating.

This is how I think the alpha version looks:

![image](https://user-images.githubusercontent.com/13920678/188201304-24d1716f-4bb6-4108-8f42-58e717155c55.png)

![image](https://user-images.githubusercontent.com/13920678/188201343-e4db4eda-2b58-4053-b0b7-b905fb91ad47.png)

![image](https://user-images.githubusercontent.com/13920678/188201326-9fbb4982-5745-4612-a023-446dc2e04c45.png)

**Comment by timbze** [comment]
@agrogers I guess some of that is still in PR... I don't have those fancy features yet beside the boldening of search result words 😄

**Comment by Noitarud** [comment]
@timbze No I don't (I did try Chinese but that's probably not what you meant) but I have noticed things that apparently others haven't… for example the concept of calling Hebrew words "feminine" (the suffix) seems rather wrong (they are "products" - just as woman was from man, a unit or element), and noted the language has other suffixes (eg, resh, but I have not got all narrowed down though).

As for Greek, my second last was Ana, they wrote "up" and you see many Bibles translations presume it is "upwards" and add "up" (and "down" for kata) when they find such a prefixed word, I was surprised it was a different kind of up, retired now in English, it is causative. You still see it sometimes, but you assume it means upwards because we are not familiar with it, eg Start Up (or astart). And another, uproar, the older definition was "throw into confusion".
Therefore a possibility for G313 is "conceive"…

Some of the words were translated with no clear consensus (which was what got me going on this, proverbs translation), words with obviously different spellings translated to be the same (see 1pe5:7, zec 10.2 (pontentially 3× same!), Luke10:19, job12:3-20 (3× understanding), twice in Pro26:11), and reversioning makes it worse (vow becomes oath becomes 'tell the truth') and they may reword other's translations (NAB 2ki 4:42 can you guess which translation they read?), some translated with no thought and copied by other versions G431 (originated with Tyndale?), some verses they have no idea so they improvise eg Lev20:17 - sometimes I think they translate their opinions of it (I am disinclined to read the heavenly scenes in Ezekiel). And those 25 opression words? The best one of the few I have to tell was H3238 to be dispossession.

English words (nouns verbs) do not have that many alternate meanings so theirs shouldn't either. I am suspecting I am missing something when it comes to Hebrew "face" and suspect there could be omitted particles that will refine it (or them). My biggest hurdle are the rarer words of course (thats when I am tempted to digress into looking at structure, sometimes it is profitable, I find origins and similar words, other times all I have is some clues).

Sometimes a different set of eyes notice things the first does not (eg G527 word relates to "salt", suspect alluding to growth of salt crystals, word should be "burgeoning", or if wanting to be thoeretically accurate, "precipitating" might work depending on whether there are other possible uses). Sometimes I get to see a few authors disagreeing with consensus, G4352 really is homage not worship.

That is all I'll share for now…
(I should qualify some more of this)

**Comment by agrogers** [comment]
> @agrogers I guess some of that is still in PR... I don't have those fancy features yet beside the boldening of search result words 😄

I use my own build which is here: https://github.com/AndBible/and-bible/tree/Study-Bible-2
I forget soemtimes what is in my version and the released versions.

**Comment by Noitarud** [comment]
This is my present version of Mk11:22³/₃-25:

> Have faith of God; for certainly I express to-you-all, because any (of-you) say to-this mountain lift and toss unto the sea and not deliberate in their heart but believing because their expressions become, it will-be whether (they) say; through this [I] express to-you-all, every extent any[body] asks praying, believing because of getting, and to-you-all [it] will-be; and when standing praying, dismiss if having some[thing] acting-on some[one], would also your Father :occupying heaven dismiss to you-all your offences.

( ) are a subject I had to select, : are double words made one.
this is using the KJV (esword) numbers, if I use W&H the result differs (which I am disappointed about since there should be only one original text).

**Comment by Noitarud** [comment]
### its not updating!
@agrogers (sorry meant @timbze) I get "app not installed" do I have to uninstall previous? I did not update when told to for #2143 and thought to check that too (I just updated the last post to include his name; looks like I might have tried updating around that time and failed).

**Comment by agrogers** [comment]
Hi @Noitarud . My branch is my own build that also includes some database changes. So i am not sure you want to use it as your main app. If you do, I think it should be able to be installed alongside the public releases. You can grab my apk from here: https://drive.google.com/drive/folders/1m1oiNDEi4_1uJrjhRP9aQr1xS56fEPQR?usp=sharing 

Be sure to backup your data before. It should restore fine into my version. And my version has 'Debug' in the name.

**Comment by Noitarud** [comment]
@agrogers > Hi @Noitarud . My branch is my own build that also includes some database changes. So i am not sure you want to use it as your main app. If you do, I think it should be able to be installed alongside the public releases.…

It is recognised as an update, still same error as before. Is there a solution or do I have to uninstall?

Are the db changes functionally superior? Is the build tailored to some benefit? Just asking so I could get the most out of it!

**Comment by Noitarud** [comment]
@agrogers bump (I thought it would note you when I pressed quote, I then added the name)

**Comment by agrogers** [comment]
Sorry @Noitarud . I am not exactly sure what you need to do. The easiest way would be 
1. Backup your database (and keep it safe!)
2. Uninstall the publicly available version
3. Install my version
4. Restore the database.

Remember that you might have troubles restoring the database from my version back into the public version. So proceed with caution.

**Comment by Noitarud** [comment]
@agrogers, does this mean you will need two backup options, one for the public version and one for your version?😉

**Comment by Noitarud** [comment]
@agrogers, does this mean you will need two backup options, one for the public version and one for your version?😉

Is your version tailored to some benefit?

**Comment by agrogers** [comment]
Hi. I'm not sure how to answer that. I have my own build in which I include
all my changes like the search stats and color buttons. And it does include
database changes. So my backup includes those changes. I haven't tried to
restore my database back into a standard installation. I assume I would
have problems. I definitely have troubles running the standard version
against my database.

**Comment by tuomas2** [comment]
As it has database changes it can't be used with standard version.
Standard versions's backup can be used with @agrogers version.

As with alpha and beta versions in general, I recommend @agrogers version to be used with extra caution, to provide alpha-stage testing feedback to him.

**Comment by Noitarud** [comment]
If I cannot have both installed at the same time, it will be difficult to choose to migrate.
Also @agrogers, I think I have found a hint to why it was not installing, they supply multiple kinds, so downloaded (early 1/2023):
- Duckduckgo-5.145.5-play-release.apk
- Com.duckduckgo.mobile.android_51454000.apk

The Play one failed with the same message as yours. Otherwise I do not know what the problem is.
Edit 2024/1: Ghost commander author (zc2) said about repositories: _There's only one thing for sure, you cannot mix apps and plugins from different sources, since the signatures won't match._

**Comment by Noitarud** [comment]
@Timbze I just downloaded 684, its not there, is this feature still in alpha test phase only?

**Comment by timbze** [comment]
> @timbze I just downloaded 684, its not there, is this feature still in alpha test phase only?

Yes, it's still in test phase, but it is in beta now. You can get it by getting in Beta on Play store, or by downloading the apk on Github releases, if you'd like to do one of them.

**Comment by Noitarud** [comment]
@timbze I obviously do not have issue reopening authority, I just discovered 719β does not make bold when you enter "strong:H1234". By the way the beta sucks too much (see the study pad bug of mine).

**Comment by Noitarud** [comment]
Another minor but possibly convenient bug is you do one search and select a result, then in another window do a search and press back or select, then in first window press back and you get first's results again but with the second's words highlighted.

**Comment by tuomas2** [comment]
> Another minor but possibly convenient bug is you do one search and select a result, then in another window do a search and press back or select, then in first window press back and you get first's results again but with the second's words highlighted.

You better create another bug ticket about that.
